### ISSUE TITLE
Docker 1.7.0 volumes set to false on upgrade

### ISSUE THREAD

**Issue by tomcon21** [docker/docker#14347]
Hi, 

We have recently upgraded the components of out multi-container logging service to the latest version. In doing so, we decided to upgrade to docker version: 1.7.0. 

This all seemed to go smoothly until I tested the restart policies for our containers by rebooting the lab box. All the containers restarted successfully except for Elasticsearch, which would unsuccessfully attempt to restart over and over.

We began by rolling back the version of Elasticsearch but the problem persisted, so started to rule out other components of the service. In the end the problem was with Docker 1.7.0, we discovered this by downgraded to 1.6.2 and the restart policies began to work again after reboot. 

Our current set-up consists of the following:
ELK stack with Supervisor running and monitoring the processes in each container
A Data-Only container which is a volume mounted on the host for ES data

I am wondering has anyone else encountered this problem?

**Comment by runcom** [comment]
@tomcon21 which restart policity the elastisearch container has?

**Comment by tomcon21** [comment]
docker run -d -p 9200:9200 -p 9300:9300 --name elasticsearch --restart=always --volumes-from elkdata elk:elasticsearch

**Comment by cpuguy83** [comment]
@tomcon21 Can you provide logs from the daemon for when these restarts fail?

**Comment by tomcon21** [comment]
2015-07-02 13:45:50,192 INFO supervisord started with pid 1
Traceback (most recent call last):
  File "/usr/bin/supervisord", line 9, in <module>
    load_entry_point('supervisor==3.0b2', 'console_scripts', 'supervisord')()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 360, in main
    go(options)
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 370, in go
    d.main()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 83, in main
    self.run()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 100, in run
    self.runforever()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 252, in runforever
    [ group.transition() for group  in pgroups ]
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 697, in transition
    proc.transition()
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 541, in transition
    self.spawn()
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 219, in spawn
    self.dispatchers, self.pipes = self.config.make_dispatchers(self)
  File "/usr/lib/python2.7/dist-packages/supervisor/options.py", line 1676, in make_dispatchers
    dispatchers[stdout_fd] = POutputDispatcher(proc, etype, stdout_fd)
  File "/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py", line 98, in __init__
    backups=backups)
  File "/usr/lib/python2.7/dist-packages/supervisor/options.py", line 445, in getLogger
    backups, stdout)
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 344, in getLogger
    handlers.append(RotatingFileHandler(filename,'a',maxbytes,backups))
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 172, in __init__
    FileHandler.__init__(self, filename, mode)
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 98, in __init__
    self.stream = open(filename, mode)
IOError: [Errno 30] Read-only file system: '/var/log/supervisor/elasticsearch/elasticsearch.log'

**Comment by cpuguy83** [comment]
@tomcon21 So you are running supervisord in the container, and supervisord can't start elasticsearch?

Is `/var/log/supervisor/elasticsearch/elasticsearch.log` being mounted in from a volume?
It's strange that this is being set to read-only.

Also if you can provide `docker info` and `docker version` outputs, that'd be great.

**Comment by tomcon21** [comment]
docker info
Containers: 6
Images: 67
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 249
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-32-generic
Operating System: Ubuntu precise (12.04.5 LTS)
CPUs: 2
Total Memory: 1.955 GiB

 docker version
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64

**Comment by cpuguy83** [comment]
@tomcon21 Thanks, can you answer my question about your log dir?
Maybe you can just paste the inspect output from that container.

**Comment by tomcon21** [comment]
@cpuguy83  sorry, 

Below is the output from 1.6.2 
 docker version
Client version: 1.6.2
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 7c8fca2
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 7c8fca2
OS/Arch (server): linux/amd64

Docker inspect elasticsearch Output: 

    "VolumesRW": {
        "/elasticsearch_data": true,
        "/opt/elk/elasticsearch/config/templates": true,
        "/var/log/supervisor/cron": true,
        "/var/log/supervisor/elasticsearch": true
    }

Below is the output from 1.7.0

docker version
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64


"VolumesRW": {
        "/elasticsearch_data": false,
        "/opt/elk/elasticsearch/config/templates": false,
        "/var/log/supervisor/cron": false,
        "/var/log/supervisor/elasticsearch": false
    }

**Comment by cpuguy83** [comment]
Great, thank you, this looks like a regression in the volumes rewrite.
Tagging for 1.7.1.

**Comment by cpuguy83** [comment]
If you can provide the  the `Volumes` field as well, that would be great.

**Comment by calavera** [comment]
@tomcon21 it would also be really helpful if you could provide the `docker run` command you use to start those containers.

**Comment by calavera** [comment]
oh nevermind, I just saw it

**Comment by calavera** [comment]
@tomcon21 having the volumes information from the parent container would also help. I have the feeling that the issue is caused by that `--volumes-from`.

**Comment by calavera** [comment]
Unfortunately, I cannot reproduce this issue. Upgrading from 1.6.2 to 1.7 works in my tests. We're going to keep this issue in the milestone but we're not going to block the RC release for it.

@tomcon21 the more information you can provide to reproduce this issue the better, thanks.

**Comment by tomcon21** [comment]
@cpuguy83  

Below is the output from 1.7.0:

"Volumes": {
        "/elasticsearch_data": "/var/lib/docker/volumes/950b62496b67059b69d41b0701bd2561949a04276e9242dc9312ef955b125ca6/_data",
        "/opt/elk/elasticsearch/config/templates": "/var/lib/docker/volumes/2d532aef71e0d6e795bd06920e73d498fd4667368a9ea27af13a3099fe95ffcd/_data",
        "/var/log/supervisor/cron": "/var/lib/docker/volumes/911e840fa422dc28aa07a28bd5032d3bb396d432a6fd1e9208adfdbb8a53b5d8/_data",
        "/var/log/supervisor/elasticsearch": "/var/lib/docker/volumes/67049a1d65854337c096767d952a66b1ad210a7d33d1e5feee76b12d86cdfffa/_data"
    },
    "VolumesRW": {
        "/elasticsearch_data": false,
        "/opt/elk/elasticsearch/config/templates": false,
        "/var/log/supervisor/cron": false,
        "/var/log/supervisor/elasticsearch": false
    },



Below is the output from 1.6.2:

    "Volumes": {
        "/elasticsearch_data": "/var/lib/docker/vfs/dir/87bd9ed5b6606920850cb214d3f67d337603cf50e91aec096c001965872c1e5d",
        "/opt/elk/elasticsearch/config/templates": "/var/lib/docker/vfs/dir/fe066eb3e32d5630b34e9d1aebcef17bc8f3d6383e84c341f301b6ffa7cc465d",
        "/var/log/supervisor/cron": "/var/lib/docker/vfs/dir/30149cfbaac2c267ae73ac8103dc71d30467180b619b88f52a2c9434ca6584c6",
        "/var/log/supervisor/elasticsearch": "/var/lib/docker/vfs/dir/0475bdab6be570c8bb1f815cd9e74b7b24f4826029d7de5d4eee2283bf1de6e1"
    },
    "VolumesRW": {
        "/elasticsearch_data": true,
        "/opt/elk/elasticsearch/config/templates": true,
        "/var/log/supervisor/cron": true,
        "/var/log/supervisor/elasticsearch": true
    }

**Comment by tomcon21** [comment]
@calavera  

"Volumes": {
        "/elasticsearch_data": "/var/lib/docker/vfs/dir/87bd9ed5b6606920850cb214d3f67d337603cf50e91aec096c001965872c1e5d"
    },
    "VolumesRW": {
        "/elasticsearch_data": true
    }

**Comment by calavera** [comment]
@tomcon21 do you mind to post the instruction you use to create the volumes in the parent container? I'm still trying to figure out how that can happen.

**Comment by calavera** [comment]
@tomcon21 have you tried to start another container inheriting the volumes from that same container to see if the output is the same? That would also be super helpful

**Comment by tomcon21** [comment]
@calavera  

ES Container:
VOLUME ["/opt/elk/elasticsearch/config/templates", "/var/log/supervisor/elasticsearch", "/var/log/supervisor/cron"]

Here is the Dockerfile for my Data Container:

FROM ubuntu:14.04
RUN mkdir -p /elasticsearch_data && touch /elasticsearch_data/elkstack_dummy_file
RUN useradd elasticsearch --password elasticsearchpwd --shell /bin/bash
RUN chown -R elasticsearch:elasticsearch /elasticsearch_data

VOLUME ["/elasticsearch_data"]

----------------
Run command for Data Only container:
docker run --name elkdata -v /elasticsearch_data elkstack:elkdata echo "Data-only container for elasticsearch"

------------------
Do you want me to spin up a container that uses the data containers volume?

**Comment by tomcon21** [comment]
@calavera

How I encountered this error was by removing Docker with all associated volumes, images and containers from the lab machine. 

I then installed 1.7.0 , built my images and then ran the containers. I performed a reboot and Elasticsearch would not restart.

Since then I performed an upgrade without removing Docker or any of the volumes, images and containers from the lab machine and it seemed to work.

Docker version 1.7.0, build 0baf609

This is now the output : 
"VolumesRW": {
        "/elasticsearch_data": true,
        "/opt/elk/elasticsearch/config/templates": true,
        "/var/log/supervisor/cron": true,
        "/var/log/supervisor/elasticsearch": true
}

  "Volumes": {
        "/elasticsearch_data": "/var/lib/docker/volumes/7a1065a833848f6be01e8a1bb68593b17b19e7e25ba72f4a894ac0f0e56e8d40/_data"
    },
    "VolumesRW": {
        "/elasticsearch_data": true
}

It seems to be this scenario that triggers this behavior. Our requirement would require a clean install.
